Assets/MixedRealityToolkit-SDK/Features/Input/Handlers/DragAndDropHandler.cs

```diff
@@ -150,7 +150,7 @@ void IMixedRealityPointerHandler.OnPointerDown(MixedRealityPointerEventData even
                     ? focusDetails.Point
                     : hostTransform.position;
 
-            isDraggingWithSourcePose = currentPointer == MixedRealityToolkit.InputSystem.GazeProvider.GazePointer;
+            isDraggingWithSourcePose = currentPointer.PointerId == MixedRealityToolkit.InputSystem.GazeProvider.GazePointer.PointerId;
```

Comment:
What's the difference?

Reply:
Better equality check because we're using the interface to check equality and not the actual class implementation (Which does this under the hood)